extensions/pyRevitTools.extension/checks/audit_all_check.py

```diff
@@ -0,0 +1,925 @@
+        self.text_notes_count = len(text_notes)
+        self.text_notes_types_count = len(text_notes_types)
+        self.text_notes_types_wf_count = (
+            cnt.count_textnote_types_with_changed_width_factor(text_notes_types)
+        )
+        self.text_bg_count = cnt.count_textnote_types_with_opaque_background(text_notes_types)
+        self.text_notes_caps_count = cnt.count_text_notes_with_all_caps(text_notes)
+
+
+class GroupInfo:
+    """Handles group-related data."""
+
+    def __init__(self, document):
+        self.detail_groups_instances_count = cnt.count_detail_group_instances(document)
+        self.detail_groups_types_count = cnt.count_detail_groups_types(document)
+        self.model_groups_instances_count = cnt.count_model_group_instances(document)
+        self.model_groups_types_count = cnt.count_model_groups_types(document)
+
+
+class ReferencePlaneInfo:
+    """Manages reference plane metrics."""
+
+    def __init__(self, document):
+        reference_planes = q.get_elements_by_class(DB.ReferencePlane, doc=document)
+        self.reference_planes_count = len(reference_planes)
+        self.unnamed_ref_planes_count = cnt.count_unnamed_reference_planes(
+            reference_planes
+        )
+
+
+class DimensionsInfo:
+    """Handles dimension-related data."""
+
+    def __init__(self, document):
+        dimensions = [
+            d
+            for d in q.get_elements_by_categories(
+                [DB.BuiltInCategory.OST_Dimensions], doc=document
+            )
+            if d.OwnerViewId and d.ViewSpecific and d.View
+        ]
+        dimensions_types = (
+            DB.FilteredElementCollector(document)
+            .OfClass(DB.DimensionType)
+            .WhereElementIsElementType()
+            .ToElements()
+        )
+        self.dim_types_count = len(dimensions_types)
+        self.dim_count = len(dimensions)
+        self.dim_overrides_count = cnt.count_dimension_overrides(dimensions)
+
+
+class RevisionsInfo:
+    """Manages revision clouds count."""
+
+    def __init__(self, document):
+        self.revision_clouds_count = cnt.count_revision_clouds(document)
+
+
+class ReportData:
+    """
+    ReportData class is responsible for collecting and organizing various pieces of information
+    about a Revit document and its associated metadata, project information, element counts, warnings,
+    links, families, text notes, groups, reference planes, dimensions, and revisions.
+
+    Attributes:
+        _KEY_MAPPING (OrderedDict): A mapping of keys to attribute paths for various pieces of information.
+
+    Methods:
+        __init__(self, document=None):
+            Initializes the ReportData instance with the provided Revit document.
+        COLUMNS(self):
+            Returns a list of column names based on the keys in _KEY_MAPPING.
+        to_list(self):
+        export_to_csv(self, export_file_path=EXPORT_FILE_PATH, headers=None):
+    """
+
+    _KEY_MAPPING = OrderedDict(

```

Comment:
this could be a module-level constant, but since you use the class only once it is not that bad to leave it here.